packages/millicast-sdk/src/utils/Codecs.js

```diff
@@ -372,24 +382,50 @@ function extractSEIPayload (rbsp) {
   }
 }
 
+function resolveUnregisteredMessageType (uuid) {
+  const timecodeUuid = new Uint8Array(parseUUID(DOLBY_SEI_TIMESTAMP_UUID))
+  const legacySdkUuid = new Uint8Array(parseUUID(DOLBY_SEI_DATA_UUID))
+  const newSdkUuid = new Uint8Array(parseUUID(DOLBY_SDK_TIMESTAMP_UUID))
+
+  if (timecodeUuid.every((value, index) => value === uuid[index])) return UNREGISTERED_MESSAGE_TYPE.TIMECODE
+  if (legacySdkUuid.every((value, index) => value === uuid[index])) return UNREGISTERED_MESSAGE_TYPE.LEGACY
+  if (newSdkUuid.every((value, index) => value === uuid[index])) return UNREGISTERED_MESSAGE_TYPE.NEW
+}
+
 function getSeiUserUnregisteredData (metadata, payloadContent) {
   let idx = 0
   metadata.uuid = payloadContent.subarray(idx, idx + 16)
   idx += 16
+
+  // There are 3 possible pathways here
+  // the old UUID - where-in there is no timecode in the body
+  // the new UUID - where-in there is timecode in the body
+  // AMF/KLV timestamps - where there is no other data to parse
+  const messageType = resolveUnregisteredMessageType(metadata.uuid)
   const content = payloadContent.subarray(idx)
-  if (isUUIDTimestamp(metadata.uuid)) {
-    metadata.timecode = convertSEITimestamp(content)
-  } else {
-    metadata.unregistered = content
+  switch (messageType) {
+    case UNREGISTERED_MESSAGE_TYPE.LEGACY :
+      metadata.unregistered = content
+      break
+    case UNREGISTERED_MESSAGE_TYPE.TIMECODE:
+      metadata.timecode = convertSEITimestamp(content)
+      break
+    case UNREGISTERED_MESSAGE_TYPE.NEW:
+    {
```

Comment:
```suggestion
```
Do we need this bracket?

Reply:
Oh I didn't have them, but lint didn't like em, so added them back in